initPTB.m

```diff
@@ -152,3 +120,122 @@
 
 
 end
+
+
+function initDebug(cfg)
+
+% init PTB with different options in concordance to the debug Parameters
+Screen('Preference','SkipSyncTests', 0);
+if cfg.debug
+    
+    Screen('Preference', 'SkipSyncTests', 2);
+    Screen('Preference', 'Verbosity', 0);
+    Screen('Preferences', 'SuppressAllWarnings', 2);
+    
+    fprintf('\n\n\n\n')
+    fprintf('########################################\n')
+    fprintf('##   DEBUG MODE. TIMING WILL BE OFF.  ##\n')
+    fprintf('########################################')
+    fprintf('\n\n\n\n')
+    
+    testKeyboards(cfg)
+    
+end
+
+if cfg.testingTranspScreen
+    PsychDebugWindowConfiguration
+end
+
+end
+
+function initKeyboard
+
+% Make sure keyboard mapping is the same on all supported operating systems
+% Apple MacOS/X, MS-Windows and GNU/Linux:
+KbName('UnifyKeyNames');
+
+% Don't echo keypresses to Matlab window
+ListenChar(-1);
+
+end
+
+function cfg = initAudio(cfg)
+    
+    if cfg.initAudio
+        
+        requestedLatency = 3;
+        
+        InitializePsychSound(1);
+
+        cfg.audio.devIdx= [];
+        cfg.audio.playbackMode = 1;
+            
+        if IsWin
```

Comment:
Simplified this `if` block but I think I can do better.

The issue here is that you want to use a specific sound card not that you run on windows so it would make more sense (and would make the whole thing more general) to check for a field in cfg asking to use a specific device with some specific parameters.